test/external/ringtest/CMakeLists.txt

```diff
@@ -41,6 +41,38 @@ foreach(processor cpu gpu)
     set(gpu_arg)
     set(special_gpu_arg)
   endif()
+  if("${processor}" STREQUAL "cpu")
```

Comment:
Why not just move this outside the `foreach(processor ...)` loop if it doesn't work on GPU (is there an issue for that?).